cornac/models/narre/recom_narre.py

```diff
@@ -172,14 +183,20 @@ def _fit_narre(self):
         import tensorflow as tf
         from tensorflow import keras
         from .narre import get_data
+        from ...eval_methods.base_method import rating_eval
+        from ...metrics import MSE
         loss = keras.losses.MeanSquaredError()
-        if self.optimizer == 'rmsprop':
-            optimizer = keras.optimizers.RMSprop(learning_rate=self.learning_rate)
-        else:
-            optimizer = keras.optimizers.Adam(learning_rate=self.learning_rate)
+        if not hasattr(self, 'optimizer_'):
+            if self.optimizer == 'rmsprop':
+                self.optimizer_ = keras.optimizers.RMSprop(learning_rate=self.learning_rate)
+            else:
```

Comment:
should check the correct `self.optimizer == adam`, and `raise ValueError` otherwise to avoid any assumptions